x-pack/plugins/profiling/server/routes/query.ts

```diff
@@ -50,11 +50,6 @@ export function autoHistogramSumCountOnGroupByField(searchField: string, topNIte
       group_by: {
         terms: {
           field: searchField,
-          // We remove the ordering since we will rely directly on the natural
```

Comment:
While I totally agree with reducing the number of displayed traces to 10, I have two concerns here.

1. I dislike relying on default values. They can change in the future with unknown rationale to unforeseeable values, not matching with our intent. So if we decide that 10 is good - let's hard-code '10' here.

2. To make the comment superfluous, we should order the results by 'Sum of Count' and not rely on some inexact assumptions. Staying with the "good enough" approach may bite us later - there may pop up corner cases where this may lead to very confusing or misleading output.

Reply:
If adding
```
      count: {
        sum: {
          field: 'Count'
        }
      }
```
on the same level as 'group_by', then this represents the total count per bucket.
So we can completely ignore 'doc_count' further on.
The bucket items then need to be reordered by 'count' (they will be returned order by 'doc_count').

To test in the Kibana dev console (WARNING: there is no filter - use it with care)
```
GET profiling-events-all/_search
{
  "size": 0,
  "aggs": {
    "histogram": {
      "auto_date_histogram": {
        "field": "@timestamp",
        "buckets": 50
      },
      "aggs": {
        "group_by": {
          "terms": {
            "field": "ThreadName",
            "size": 10,
            "execution_hint": "map"
          },
          "aggs": {
            "count": {
              "sum": {
                "field": "Count"
              }
            }
          }
        },
        "count": {
          "sum": {
            "field": "Count"
          }
        }
      }
    }
  }
}
```

Reply:
Thanks! I use explicit ordering for the primary aggregation (`count.value` descending). I also added the `sum` aggregation as suggested and stopped using `doc_count`.

Reply:
Should we also display 'other'? (By subtracting the sum of all buckets of the total count)

Reply:
I agree this needs to be done, but I created a [separate issue](https://github.com/elastic/prodfiler/issues/2474) since there are two possible implementations for this. I'd prefer to handle it separately so that we can fix the outstanding performance problem first.